### PR TITLE
Port remaining casefiddle Lisp functions

### PR DESCRIPTION
**Pull request by nickdrozd**
A lot of work is still being done in C, but all the exposed Lisp stuff is in Rust.

### CODE REVIEW

File: src/casefiddle.c [hunk 1]
```diff
@@ -482,89 +467,17 @@ casify_region (enum case_action flag, Lisp_Object b, Lisp_Object e)
...
-  Lisp_Object bounds = Qnil;
-
-  if (!NILP (region_noncontiguous_p))
-    {
-      bounds = call1 (Fsymbol_value (intern ("region-extract-function")),
-		      intern ("bounds"));
-
-      while (CONSP (bounds))
-	{
-	  casify_region (CASE_UP, XCAR (XCAR (bounds)), XCDR (XCAR (bounds)));
```
Comment (Wilfred) [inline src/casefiddle.c]: I'm not actually sure this C code was correct. It calls XCAR twice, but it's only checked that the outer value is a cons cell. Not that it's a problem for us :)

File: rust_src/src/casefiddle.rs [hunk 2]
```diff
@@ -36,4 +96,132 @@ pub fn upcase(object: LispObject) -> LispObject {
...
+    if region_noncontiguous_p.is_nil() {
+        unsafe { casify_region_nil(action, beg, end) }
+    } else {
+        let mut bounds = call!(
+            symbol_value(intern("region-extract-function")),
+            intern("bounds")
+        );
+
+        while bounds.is_cons() {
+            let cons = bounds.as_cons_or_error();
```
Comment (Wilfred) [inline rust_src/src/casefiddle.rs]: We've just checked that it is a cons, do we need the or_error version here?

Comment (brotzeit) [reply rust_src/src/casefiddle.rs]: Maybe something like `while let Some(c) = bounds.as_cons()`.

Comment (nickdrozd) [reply rust_src/src/casefiddle.rs]: I thought something was ugly about that. Updated.

**Pull request merged**